benchmarking-tests/benchmark_utils.py

```diff
@@ -63,11 +71,22 @@ def run_gather_metrics(args):
             if change['old_val'] is None:
                 block_num_transactions = len(change['new_val']['block']['transactions'])
                 time_now = time.time()
-                logger.info('{} {}'.format(time_now, block_num_transactions))
                 csv_writer.writerow([str(time_now), str(block_num_transactions)])
 
-                num_transactions -= block_num_transactions
-                if num_transactions == 0:
+                # log statistics
+                if initial_time is None:
+                    initial_time = time_now
+
+                num_transactions_received += block_num_transactions
+                elapsed_time = time_now - initial_time
+                elapsed_time = elapsed_time if elapsed_time != 0 else 1
```

Comment:
This is going to make the first `transactions_per_second` equal to the number of transactions in the first block, which is misleading. What to do instead? Maybe arrange things so that the first `transactions_per_second` is `float('nan')` (i.e. Not a Number)? http://stackoverflow.com/questions/19374254/assigning-a-variable-nan-in-python-without-numpy